core/lib/spree/core/product_duplicator.rb

```diff
@@ -70,5 +70,12 @@ def reset_properties
         end
       end
     end
+
+    def generate_new_sku(sku)
+      loop do
+        sku = "COPY OF #{sku}"
+        break sku unless Spree::Variant.where(sku: sku).first
```

Comment:
Thanks @mtomov I must add unit test as well.

It get out the loop if no variant found with the new sku value.